controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -110,49 +109,20 @@ class ControllerInterface : public rclcpp_lifecycle::node_interfaces::LifecycleN
   template <typename ParameterT>
   auto auto_declare(const std::string & name, const ParameterT & default_value)
   {
-    if (!node_->has_parameter(name))
+    if (!lifecycle_node_->has_parameter(name))
     {
-      return node_->declare_parameter<ParameterT>(name, default_value);
+      return lifecycle_node_->declare_parameter<ParameterT>(name, default_value);
     }
     else
     {
-      return node_->get_parameter(name).get_value<ParameterT>();
+      return lifecycle_node_->get_parameter(name).get_value<ParameterT>();
     }
   }
 
-  /**
-   * The methods below are a substitute to the LifecycleNode methods with the same name.
-   * The Life cycle is shown in ROS2 design document:
-   * https://design.ros2.org/articles/node_lifecycle.html
-   * We cannot use a LifecycleNode because it would expose change-state services to the rest
-   * of the ROS system.
-   * Only the Controller Manager should have possibility to change state of a controller.
-   *
-   * Hopefully in the future we can use a LifecycleNode where we disable modifications from the outside.
-   */
-  CONTROLLER_INTERFACE_PUBLIC
-  const rclcpp_lifecycle::State & configure();
-
-  CONTROLLER_INTERFACE_PUBLIC
-  const rclcpp_lifecycle::State & cleanup();
-
-  CONTROLLER_INTERFACE_PUBLIC
-  const rclcpp_lifecycle::State & deactivate();
-
-  CONTROLLER_INTERFACE_PUBLIC
-  const rclcpp_lifecycle::State & activate();
-
-  CONTROLLER_INTERFACE_PUBLIC
-  const rclcpp_lifecycle::State & shutdown();
-
-  CONTROLLER_INTERFACE_PUBLIC
-  const rclcpp_lifecycle::State & get_state() const;
```

Comment:
can't we use `get_state` still but just forward lifecycne_node_->get_state() from it?

Reply:
Yep, that's possible